jobs.groovy.j2

```diff
@@ -289,14 +307,36 @@ def build_publishers(v) {
     ( 'shell', others )
     currently only shell has been implemented.
 
-    params v: dictionary containing the job values */
-def build_steps(v) {
+    params job: dictionary the job steps (``with_steps``) */
+def build_steps(dash_project, dash_branch_name, job_name, job) {
     return {
-        for (_step in v) {
+        for (_step in job.with_steps) {
             if (_step.type == 'shell') {
                 shell(_step.cli.join("\n") + "\n")
             }
         }
+
+        /* not every job produces an artifact, so make sure we
+           don't try to fetch artifacts for jobs that don't
+           produce them */
+        if (job.archive_artifacts) {
+            for (artifact in job.archive_artifacts) {
+                copyArtifacts(
+                    "${dash_project}/${dash_branch_name}/${job_name}") {
```

Comment:
:question: This looks like it would be a common pattern, could it be a function?